### PR TITLE
fix: 582memory store authentication error code

### PR DESCRIPTION
**Pull request by vineet-garg**
## Related issue
#582 
## Proposed changes

Change storage memory error to what ResourceOwnerPasswordCredentialsGrantHandler is expecting

## Checklist

- [ x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ x] I have read the [security policy](../security/policy).
- [ x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: storage/memory.go [hunk 1]
```diff
@@ -352,7 +351,7 @@ func (s *MemoryStore) Authenticate(_ context.Context, name string, secret string
 		return fosite.ErrNotFound
 	}
 	if rel.Password != secret {
-		return errors.New("Invalid credentials")
+		return fosite.ErrNotFound
```
Comment (aeneasr) [inline storage/memory.go:354]: This store is really just a reference. I think it’s user friendly to retutn an understandable error message here. Not sure if there is value in making this into a generic „not found“ message

Comment (vineet-garg) [reply storage/memory.go:354]: To provide some context, the code that is calling the storage i.e. (ResourceOwnerPasswordCredentialsGrantHandler) (excerpt pasted below) expects only ErrNotFound from implementations of storage and injects its own top-level message. Earlier the calling code was propagating the error as 500 server error. I am assuming changing the calling code to react to a more appropriate error constant will break the contract with other implementations of storage and that is not an option.

} else if err := c.ResourceOwnerPasswordCredentialsGrantStorage.Authenticate(ctx, username, password); errors.Is(err, fosite.**ErrNotFound**) {
		return errorsx.WithStack(fosite.ErrInvalidGrant.WithHint("Unable to authenticate the provided username and password credentials.").WithWrap(err).WithDebug(err.Error()))

Comment (vineet-garg) [reply storage/memory.go:354]: @aeneasr Please let me know if you have another way to solve this or you prefer to leave it the way it is. Thanks.

Comment (aeneasr) [reply storage/memory.go:354]: Hey - totally my bad, I forgot to reply here! I think we have a "WithDebug" helper on fosite errors which we could use here to provide more context, so:

```suggestion
		return fosite.ErrNotFound.WithDebug("Invalid credentials")
```

**Comment by vineet-garg** [comment]
@aeneasr Is Anyone planning to fix the failures separately? The are unrelated to my PR.

**Comment by aeneasr** [comment]
Yes!

**Review by aeneasr — APPROVED** [review]

**Comment by aeneasr** [comment]
TY :)

**Pull request merged**